### PR TITLE
[`feat`] Implementing threshold consistent margin loss

### PR DESCRIPTION
**Pull request by ir2718**
Hi,

this PR implements the loss presented in the paper [Threshold-Consistent Margin Loss for Open-World Deep Metric Learning](https://arxiv.org/abs/2307.04047). The loss is combined with a base loss provided by the user. The default hyperparameters are chosen using the values provided in the section 5.2.

Aside from the implementation, I also added a section in the docs and a test case.

### CODE REVIEW

**Comment by KevinMusgrave** [comment]
Thank you @ir2718!

I wonder if we should get rid of the `base_loss` argument, and let the user add the two together themselves. For example, if the base loss is ContrastiveLoss then:

```python
loss_fn = ThresholdConsistentMarginLoss()
base_loss_fn = ContrastiveLoss()

# when computing loss
loss = loss_fn(embeddings, labels) + base_loss_fn(embeddings, labels)
```

And if a user really wants to combine them into a single loss, they can wrap it in a function:
```python
def total_loss():
    return  loss_fn(embeddings, labels) + base_loss_fn(embeddings, labels)
``` 

Or use the MultipleLosses wrapper:
```python
from pytorch_metric_learning.losses import MultipleLosses
loss_fn = MultipleLosses([ContrastiveLoss(), ThresholdConsistentMarginLoss()])
```

**Comment by ir2718** [comment]
Oh, I wasn't aware MultipleLosses existed. I removed the base loss as requested.

**Comment by KevinMusgrave** [comment]
Thanks @ir2718!
